stdlib/2and3/ctypes/__init__.pyi

```diff
@@ -229,16 +236,28 @@ class Array(Generic[_T], Sized, _CData):
     _type_: ClassVar[Type[_T]] = ...
     raw: bytes = ...  # TODO only available with _T == c_char
     value: bytes = ...  # TODO only available with _T == c_char
-    def __init__(self, *args: _T) -> None: ...
+    # TODO These methods cannot be annotated correctly at the moment.
+    # All of these "Any"s stand for the array's element type, but it's not possible to use _T here,
+    # because of a special feature of ctypes.
+    # By default, when accessing an element of an Array[_T], the returned object has type _T.
+    # However, when _T is a "simple type" like c_int, ctypes automatically "unboxes" the object
+    # and converts it to the corresponding Python primitive. For example, when accessing an element
+    # of an Array[c_int], a Python int object is returned, not a c_int.
+    # This behavior does *not* apply to subclasses of "simple types".
+    # If MyInt is a subclass of c_int, then accessing an element of an Array[MyInt] returns
+    # a MyInt, not an int.
+    # This special behavior is not easy to model in a stub, so for now all places where
+    # the array element type would belong are annotated with Any instead.
+    def __init__(self, *args: Any) -> None: ...
```

Comment:
TBH I don't like this since this will give a sense of false safety. `Array` is one of the most used types (at least in the internal code I have seen recently) but annotating something with `Array[c_int]` is not any better than just `Array`. I think this change should be reverted and instead the special case for wrapping/unwrapping of simple types can b written as a simple mypy plugin hook for `Array.__getitem__` etc.

Reply:
Yes, I agree the current solution is not great.

Is there any documentation on how to write mypy plugins? Nothing came up for ["plugin"](https://mypy.readthedocs.io/en/latest/search.html?q=plugin&check_keywords=yes&area=default) in the mypy docs, and ["mypy plugin"](https://duckduckgo.com/?q=mypy+plugin) on DDG just gives me plugins that integrate mypy into other tools.

Would it make sense to add the plugin in a later PR, and use the "solution" with `Any` for now? It's at least better than when there was no stub at all.

Reply:
Unfortunately, plugins are not documented yet. They are however not hard, you can just add a  method in `mypy/plugin.py` that will be invoked for e.g. `Array.__getitem__` to produce callable signature instead of using the typeshed type.

In principle we can go with `Any` now, but if we do, I don't want this to stay for long and be eventually forgotten.

I am going to accept this, but please either open an issue or PR with the plugin. (You can ping me on gitter if you want to learn more about mypy plugins.)